core/src/main/java/com/amplifyframework/core/model/query/predicate/QueryField.java

```diff
@@ -40,84 +40,89 @@ public static QueryField field(String fieldName) {
     }
 
     /**
      * Generates a new greater than comparison object to compare this field to the specified value.
      * @param value the value to be compared
+     * @param <T> Comparable data type of the field
      * @return an operation object representing the greater than condition
      */
-    public QueryPredicateOperation gt(Object value) {
-        return new QueryPredicateOperation(fieldName, new GreaterThanQueryOperator(value));
+    public <T extends Comparable<T>> QueryPredicateOperation<T> gt(T value) {
+        return new QueryPredicateOperation<>(fieldName, new GreaterThanQueryOperator<>(value));
     }
 
     /**
      * Generates a new less or equal comparison object to compare this field to the specified value.
      * @param value the value to be compared
+     * @param <T> Comparable data type of the field
      * @return an operation object representing the less or equal condition
      */
-    public QueryPredicateOperation le(Object value) {
-        return new QueryPredicateOperation(fieldName, new LessOrEqualQueryOperator(value));
+    public <T extends Comparable<T>> QueryPredicateOperation<T> le(T value) {
+        return new QueryPredicateOperation<>(fieldName, new LessOrEqualQueryOperator<>(value));
     }
 
     /**
      * Generates a new less than comparison object to compare this field to the specified value.
      * @param value the value to be compared
+     * @param <T> Comparable data type of the field
      * @return an operation object representing the less than condition
      */
-    public QueryPredicateOperation lt(Object value) {
-        return new QueryPredicateOperation(fieldName, new LessThanQueryOperator(value));
+    public <T extends Comparable<T>> QueryPredicateOperation<T> lt(T value) {
+        return new QueryPredicateOperation<>(fieldName, new LessThanQueryOperator<>(value));
     }
 
     /**
-     * Generates a new not equals comparison object to compare this field to the specified value.
+     * Generates a new between comparison object to compare this field to the specified range of values.
+     * @param start the value to be used for the start of the range
+     * @param end the value to be used for the end of the range
+     * @param <T> Comparable data type of the field
+     * @return an operation object representing the between condition
+     */
+    public <T extends Comparable<T>> QueryPredicateOperation<T> between(T start, T end) {
+        return new QueryPredicateOperation<>(fieldName, new BetweenQueryOperator<>(start, end));
+    }
+
+    /**
+     * Generates a new beginsWith comparison object to compare this field to the specified value.
      * @param value the value to be compared
-     * @return an operation object representing the not equal condition
+     * @return an operation object representing the beginsWith condition
+     */
+    public QueryPredicateOperation<String> beginsWith(String value) {
```

Comment:
This one is also a little presumptive, I think. It is true that some strings begin with the contents of other strings. e.g., `"foo"` _begins with_ `"fo"`.

However, it is also true that an ordered collection can exhibit this property. For example,
```
STATES_BY_ORDER_OF_STATEHOOD.beginsWith(Arrays.asList(DELAWARE, PENNSYLVANIA))
```
could conceivably evaluate to `true`, as could:
```
STATES_BY_ORDER_OF_STATEHOOD.beginsWith(DELAWARE)
```
Where `PENNSYLVANIA` and `DELAWARE` are instances of some user-defined `State`.

(Even though I was born in the U.S., I don't actually know this stuff. I cheated and [looked up the domain details](https://en.wikipedia.org/wiki/List_of_U.S._states_by_date_of_admission_to_the_Union), here. :0)